.circleci/config.yml

```diff
@@ -993,51 +1029,78 @@ jobs:
                 name: Nightly Changelog
                 command: |
                   set -x
+                  S3_PATH="https://s3.amazonaws.com/dl.influxdata.com/platform/nightlies/2.0"
                   CHANGELOG_FILENAME="CHANGELOG.md"
+                  COMMIT_FILE_PATH="changelog-commit.txt"
+
+                  curl -o ${CHANGELOG_FILENAME} ${S3_PATH}/${CHANGELOG_FILENAME}
+                  curl -o ${COMMIT_FILE_PATH} ${S3_PATH}/${COMMIT_FILE_PATH}
+
+                  LAST_COMMIT=$(cat $COMMIT_FILE_PATH)
+                  NEWEST_COMMIT=${CIRCLE_SHA1}
+
                   cat \<< EOF >> env_vars.sh
-                  S3_PATH="https://s3.amazonaws.com/dl.influxdata.com/platform/nightlies/2.0"
+                  S3_PATH=${S3_PATH}
                   CHANGELOG_FILENAME=${CHANGELOG_FILENAME}
+                  COMMIT_RANGE="${LAST_COMMIT}..${NEWEST_COMMIT}"
+                  COMMIT_FILE_PATH=$COMMIT_FILE_PATH
                   SCRIPT_OPTIONS="--prepend ${CHANGELOG_FILENAME}"
                   EOF
+
       - unless:
           condition: << parameters.nightly >>
           steps:
             - run:
                 name: Release Changelog
                 command: |
                   set -x
-                  LAST_RELEASE_TAG=$(git describe --abbrev=0 --tags $(git rev-list --tags=v2* --skip=1 --max-count=1))
+                  S3_PATH="https://s3.amazonaws.com/dl.influxdata.com/influxdb/releases"
+                  CHANGELOG_FILENAME="CHANGELOG-${CIRCLE_TAG}.md"
+                  COMMIT_FILE_PATH="changelog-commit.txt"
+
+                  curl -o ${COMMIT_FILE_PATH} ${S3_PATH}/${COMMIT_FILE_PATH}
+
+                  LAST_COMMIT=$(git rev-list --tags=v2* --skip=1 --max-count=1)
```

Comment:
Does this handle rc's / tests properly?

Reply:
It wouldn't, but we haven't pushed a 2.x rc tag since prior to GA. It does make sense to handle rc tags though, i'll add that

Reply:
It's fun because the rc naming convention in 2.0 has been different than in 1x. In any case, I've improved the git-fu to ignore rc releases during runs of the release changelog